src/run.ts

```diff
@@ -0,0 +1,87 @@
+import { Argv } from 'yargs';
+import { Command, Helper, OptionsHelper } from '@dojo/cli/interfaces';
+import { join, relative, resolve } from 'path';
+import * as chalk from 'chalk';
+import createDir from '@dojo/cli-create-app/createDir';
+import renderFiles from '@dojo/cli-create-app/renderFiles';
+
+import register from './register';
+import run from './run';
+
+const pkgDir: any = require('pkg-dir');
+const packagePath = pkgDir.sync(__dirname);
+
+/**
+ * @type CreateWidgetArgs
+ *
+ * Arguments that can be passed when creating a widget using the Dojo CLI
+ *
+ * @property component       Generate a Custom Element wrapper
+ * @property name            Name used for the generated widget
+ * @property styles          Path to place generated CSS files
+ * @property tests           Path to place generated test files
+ */
+export interface CreateWidgetArgs extends Argv {
+	component: boolean;
+	name: string;
+	styles: string;
+	tests: string;
+}
+
+function getDirectoryNames(args: CreateWidgetArgs, folderName: string) {
+	const dirs = [ folderName ];
+
+	!args.styles && dirs.push(`${folderName}/styles`);
+	!args.tests && dirs.push(`${folderName}/tests/unit`);
+
+	return dirs;
+}
+
+function getRenderFilesConfig(args: CreateWidgetArgs, folderName: string, styleRoot: string, testRoot: string) {
+	const files = [
+		{
+			src: join(packagePath, 'templates', 'Component.ts'),
+			dest: join(folderName, `${args.name}.ts`)
+		},
+		{
+			src: join(packagePath, 'templates', 'styles/component.m.css'),
+			dest: join(styleRoot, `${folderName}.m.css`)
+		},
+		{
+			src: join(packagePath, 'templates', 'styles/component.m.css.d.ts'),
+			dest: join(styleRoot, `${folderName}.m.css.d.ts`)
+		},
+		{
+			src: join(packagePath, 'templates', 'tests/unit/Component.ts'),
+			dest: join(testRoot, `${args.name}.ts`)
+		}
+	];
+
+	args.component && files.push({
+		src: join(packagePath, 'templates', 'createComponentElement.ts'),
+		dest: join(folderName, `create${args.name}Element.ts`)
+	});
+
+	return files;
+}
+
+export default async function(helper: Helper, args: CreateWidgetArgs) {
+	const name = args.name;
+	const folderName = `${name.charAt(0).toLowerCase()}${name.slice(1)}`;
+	const styleRoot = args.styles || `${folderName}/styles`;
+	const testRoot = args.tests || `${folderName}/tests/unit`;
+
+	console.info(chalk.underline(`Creating your new widget: ${name}\n`));
+
+	createDir(...getDirectoryNames(args, folderName));
```

Comment:
Why do we pass in the args and not the `styleRoot` and `testRoot`?

Reply:
Since we only create the default test and style directories if a user doesn't pass respective paths, we need to pass some indication of this into `getDirectoryNames`. `styleRoot` and `testRoot` are local variables that will either be user-supplied paths or default paths, so they don't provide this information (unless we inspect the string or something, which gets ugly.) Passing in `args` so we can check for the presence of user-supplied paths directly seemed best.